src/webviews/extension-side/dataviewer/kernelDataViewerDependencyImplementation.ts

```
@@ -0,0 +1,121 @@
+        const outputs = await executeSilently(kernel.session!, command);
+        const error = outputs.find((item) => item.output_type === 'error');
+        if (error) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), error.message);
+        }
+        return outputs.map((item) => item.text?.toString());
+    }
+
+    protected async getVersion(kernel: IKernel): Promise<SemVer | undefined> {
+        try {
+            const outputs = await this.execute(kernelGetPandasVersion, kernel);
+            return outputs.map((text) => (text ? parseSemVer(text.toString()) : undefined)).find((item) => item);
+        } catch (e) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), e.message);
+            return;
+        }
+    }
+
+    private async installMissingDependencies(kernel: IKernel): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        let selection: string | undefined;
+        if (kernel) {
+            selection = this.isCodeSpace
+                ? Common.install()
+                : await this.applicationShell.showErrorMessage(
+                      DataScience.pandasRequiredForViewing(),
+                      { modal: true },
+                      Common.install()
+                  );
+        } else {
+            await this.applicationShell.showErrorMessage(DataScience.pandasRequiredForViewing());
+        }
+
+        if (!kernel) {
+            return;
+        }
+
+        const command = `${kernelPackaging(kernel)} install pandas`;
+
+        if (selection === Common.install()) {
+            try {
+                await this.execute(command, kernel);
+                sendTelemetryEvent(Telemetry.UserInstalledPandas);
+            } catch (e) {
+                sendTelemetryEvent(Telemetry.FailedToInstallPandas);
```

the, error is swallowed here.
we should log the error and also send it into the telemetry function.

---

if we don't log the original error, we'll never knows what went wrong

---

Thank you, Don! This is great feedback! I will review my PR again using your perspective in these comments

---

Changed the throwing of the error to include the message: `throw new Error(DataScience.failedToInstallPandas().format(e.message));`. How does that sound?

---

Sorry should have been more specified,
this is what I'm suggesting `sendTelemetryEvent(Telemetry.UserInstalledPandas, undefined, undefined, e);`
This way we can use the same telemetry event (with a flag indicating it failed) and we have the error information as well.
Using error.message is risky as it could contain PII and the `sendTelemetryEvent` will only take the bits that won't contain PII

here's an example `sendTelemetryEvent(EventName.EXTENSION_LOAD, durations, props, ex);`